homeassistant/components/delijn/sensor.py

```diff
@@ -0,0 +1,135 @@
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_NEXT_DEPARTURE): [{
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Optional(CONF_NUMBER_OF_DEPARTURES, default=5): cv.positive_int}]
+})
+
+
+def due_in_minutes(timestamp):
+    """Get the time in minutes from a timestamp."""
+    if timestamp is None:
+        return None
+    diff = timestamp - dt_util.now()
+    return int(diff.total_seconds() / 60)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    api_key = config[CONF_API_KEY]
+    name = DEFAULT_NAME
+
+    session = async_get_clientsession(hass)
+
+    sensors = []
+    for nextpassage in config[CONF_NEXT_DEPARTURE]:
+        stop_id = nextpassage[CONF_STOP_ID]
+        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
+        line = Passages(hass.loop,
+                        stop_id,
+                        number_of_departures,
+                        api_key,
+                        session)
+        await line.get_passages()
+        if line.passages is None:
+            _LOGGER.error("No data recieved from De Lijn")
+            return
+        sensors.append(DeLijnPublicTransportSensor(line, name))
+
+    async_add_entities(sensors, True)
+
+
+class DeLijnPublicTransportSensor(Entity):
+    """Representation of a Ruter sensor."""
+
+    def __init__(self, line, name):
+        """Initialize the sensor."""
+        self.line = line
+        self._attributes = {}
+        self._name = name
+        self._state = None
+
+    async def async_update(self):
+        """Get the latest data from the De Lijn API."""
+        await self.line.get_passages()
+        if self.line.passages is None:
+            _LOGGER.error("No data recieved from De Lijn")
+            return
+        try:
+            attributes = {}
+            first = self.line.passages[0]
+            local_timezone = pytz.timezone("Europe/Brussels")
+            if first['due_at_rt'] is not None:
+                first_passage_naive = dt_util.parse_datetime(
+                    first['due_at_rt'])
+            else:
+                first_passage_naive = dt_util.parse_datetime(
+                    first['due_at_sch'])
+            first_passage_local = local_timezone.localize(first_passage_naive)
+            first_passage_utc = dt_util.as_utc(first_passage_local)
+            self._state = first_passage_utc.isoformat()
+            self._name = first['stopname']
+            attributes['stopname'] = first['stopname']
+            attributes['line_number_public'] = first['line_number_public']
+            attributes['line_transport_type'] = first['line_transport_type']
+            attributes['final_destination'] = first['final_destination']
+            attributes['due_at_schedule'] = first['due_at_sch']
+            attributes['due_at_realtime'] = first['due_at_rt']
+            attributes['due_in_min'] = first['due_in_min']
```

Comment:
Please don't include any relative times in the state attributes.

Reply:
I've updated the pydelijn package in the latest commit so it has the option to convert the API timestamps of De Lijn (which are always of timzone "Europe/Brussels") to UTC timestamps. This solves the overall issue of the use of relative times.

Reply:
Please remove due_in_min from the attributes.

Reply:
Can you please give some more feedback as to why? Because that's the exact purpose of why I built this component, to reflect in how much time the next bus/tram will pass by at a certain stop. Are you saying the difference in time should be something that's calculated in Javascript in custom Lovelace cards then instead? Just to understand the reasoning and be sure what next action I need to make on my behalf. Thanks.

Reply:
Yes. The state which is now an absolute utc timestamp, can be shown as relative time in the lovelace card. We don't allow relative times in state or state attributes, in the state machine.

Reply:
OK, removed in the latest commit. Will use some logic in the custom Lovelace card instead.